### ISSUE TITLE
[ban-types] support banning `[]`

### ISSUE THREAD

**Issue by papb** [typescript-eslint/typescript-eslint#2582]
- [X] I have tried restarting my IDE and the issue persists.
- [X] I have updated to the latest version of the packages.
- [X] I have [read the FAQ](https://github.com/typescript-eslint/typescript-eslint/blob/master/docs/getting-started/linting/FAQ.md) and my problem is not listed.

**Repro**

```JSON
{
  "rules": {
    "@typescript-eslint/ban-types": [
      "error",
      {
        "extendDefaults": true,
        "types": {
          "[]": "Don't use the empty array type `[]`, it will only allow empty arrays. Use `SomeType[]` instead.",
          "[[]]": "Don't use `[[]]`, it would only allow an array with a single element which is an empty array. Use `SomeType[][]` instead.",
          "[[[]]]": "Don't use `[[[]]]`. Use `SomeType[][][]` instead."
        }
      }
    ]
  }
}
```

```TS
type X = [];
```

```jsonc
// tsconfig.json
{
	"compilerOptions": {
		"module": "commonjs",
		"strict": true,
		"target": "es2017",
		"outDir": "./dist",
	}
}

```

**Expected Result**

The custom ban to work

**Actual Result**

No errors

**Additional Info**

None

**Versions**

| package                            | version |
| ---------------------------------- | ------- |
| `@typescript-eslint/eslint-plugin` | `3.10.1` |
| `@typescript-eslint/parser`        | `3.10.1` |
| `TypeScript`                       | `4.0.2` |
| `ESLint`                           | `7.9.0` |
| `node`                             | `12.18.0` |
